lib/xclarity_client/services/xclarity_service.rb

```diff
@@ -64,33 +66,38 @@ def get_object_with_id(ids, includeAttributes, excludeAttributes)
         build_response_with_resource_list(response, managed_resource)
       end
 
-      def get_object_with_opts(opts, resource)
-        raise "The opts cannot be empty" if opts.empty?
-        filter = ""
-
-        response = if not opts.empty?
-          if not opts.has_key? "type"
-            if opts.has_key? "filterWith"
-              filter += "?filterWith="
-              filter += "#{opts["filterWith"]}"
-
-            elsif opts.has_key? "sort"
-              filter += ",sort=" if filter != ""
-              filter += "?sort=" if filter == ""
-              filter += "#{opts["sort"]}"
-            end
-          else
-            filter += "?type=#{opts["type"]}"
-          end
-        $lxca_log.info "XclarityClient::ManagementMixin get_object_with_include", "Sending request to #{resource} resource using the following filter: #{filter}"
-        @connection.do_get(resource::BASE_URI + filter)
-        end
-
+      def get_object_with_opts(opts, resource, uri = '')
+        response = get_response_with_opts(opts, resource, uri)
         build_response_with_resource_list(response, managed_resource)
       end
 
+      def get_headers_with_opts(opts, resource)
+        get_response_with_opts(opts, resource).headers
+      end
+
       private
 
+      #
```

Comment:
Once that we aren't going to document these methods, could you please remove the comment blocks?

Reply:
Removed it, now the code is more clean.